framework/source/class/qx/core/Property.js

```diff
@@ -991,7 +1036,7 @@ qx.Bootstrap.define("qx.core.Property",
       );
 
       if (incomingValue) {
-        code.push('if(this.', store, '===value)return value;');
+        code.push('if(equ(this.', store, ',value))return value;');
       } else if (resetValue) {
```

Comment:
...not sure, should it? It worked for me.
But please write a test-case which shows the potential problem